slurp/src/main/webapp/scripts/utils.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Parse URL <pre><code>NAME.html?key1=value1&key2=value2...</code></pre> into {key1: value1, ...}.
+ * @param {string} url The URL to be parsed. 
+ * @returns {Object.<string, string>} Dictionary of the parsed URL parameters. 
+ */
+function parseUrl(url) {
+  let paramsUrl = url.substring(url.indexOf('?') + 1);
+  let paramPairs = paramsUrl.split('&');
+  
+  let params = {};
+  for (let pair of paramPairs) {
+    let splitIndex = pair.indexOf('=');
+    
+    let key = pair.substring(0, splitIndex);
+    let value = pair.substring(splitIndex+1);
+
+    params[key] = value;
+  }
+
+  return params;
+}
+
+/**
+ * Make a random 32-digit UUID. 
+ * @returns {string} 32-digit UUID as string.
+ */
+function getRandomUuid() {
+  // https://gist.github.com/6174/6062387
+  return Math.random().toString(36).substring(2, 15) 
+    + Math.random().toString(36).substring(2, 15);
```

Comment:
It's not encouraged to generate your own UUID this way, since Math.random is not actually a high-quality random number generator. Prefer using https://github.com/uuidjs/uuid instead.

Reply:
The quality of the random number doesn't really matter, they're just used as IDs for dropdowns anyway. #41 removes all need for this anyway.